### PR TITLE
Resolved routing issues from previous PR

### PR DESCRIPTION
**Pull request by guillearria**
# Description

Please include a summary of the change and a link to which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
Fixes # (issue)

## Type of change

Please delete options that are not relevant.
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change status
- [X] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?
Please describe steps taken to ensure this feature is functional and does not break other features:

Tested with CRA server and created a build. Tested locally and with two developers.

# Checklist:
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation, if relevant
- [X] I have run the app using my feature and ensured that no functionality is broken
- [ ] My changes generate no new warnings
- [ ] There are no merge conflicts

### CODE REVIEW

File: app/src/App.js [hunk 1]
```diff
@@ -1,23 +1,37 @@
...
 import SignUpView from './views/SignUpView';
 import AppView from './views/AppView'
+
 import styled from 'styled-components'
 
 const StyledApp = styled.div`
 
 `
+const PrivateRoute = ({ component: Component, ...rest }) => (
+  <Route
```
Comment (kamry-bowman) [inline app/src/App.js:14]: Nice pattern here

File: app/src/components/MemberBox.jsx [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+
+    let checkboxes = document.getElementsByName('checkbox');
+    for(var i=0, n=checkboxes.length;i<n;i++) {
+      checkboxes[i].checked = false;
+    }
+  }
+
+  render() {
+    const members = this.state.members.map((member, index) => 
+      <span key={index} style={{marginBottom: '10px',}}>
```
Comment (kamry-bowman) [inline app/src/components/MemberBox.jsx:170]: once you have IDs, you will probably want your key to be member Ids

File: app/src/components/MemberBox.jsx [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+      )
+    })
+
+    this.setState({
+      members : updatedMembers,
+    })
+  }
+
+  toggleMemberForm = () => {
+    let memberForm = document.querySelector('.addMemberForm');
```
Comment (kamry-bowman) [inline app/src/components/MemberBox.jsx:109]: it's fairly rare that you want to use querySelector in react. usually you want to update your data, and then have your data get referenced in the render. so you would have memberFormToggle on state, and then in the render you would do:
```
<Component style={{ display: this.state.memberFormToggle ? 'flex' : 'none` }} />
```
and then in your method you would so something like:
```
toggleMemberForm = () => {
    this.setState(prevState => ({ memberFormToggle: !prevState.memberFormToggle })
}
```

File: app/src/components/MemberBox.jsx [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+  }
+
+  handleChange = e => {
+    this.setState({[e.target.name]: e.target.value});
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+
+    if (this.state.name === "" || this.state.phone === "") {
```
Comment (kamry-bowman) [inline app/src/components/MemberBox.jsx:125]: same as above, look for a way to do this without using querySelector, let me know if you'd like to discuss

File: app/src/components/MemberBox.jsx [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+      document.querySelector('.formError').style.display = 'block';
+    } else {
+      document.querySelector('.formError').style.display = 'none';
+
+      let newMember = {
+        name: this.state.name, 
+        phone: parseInt(this.state.phone),
```
Comment (kamry-bowman) [inline app/src/components/MemberBox.jsx:132]: parseInt won't work for something like '334-3210', google around about phone numbers in forms

File: app/src/components/MemberBox.jsx [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+      <MemerBoxContainer>
+        <p>Your group members will appear here. Uncheck the box to remove someone from the message chain.</p>
+        
+        <MemberList>
+          {members}
+        </MemberList>
+
+        <ButtonContainer>
+          <Button
+            onClick={() => this.toggleMemberForm()} >
```
Comment (kamry-bowman) [inline app/src/components/MemberBox.jsx:191]: you could probably just do `{this.toggleMemberForm}`

File: app/src/components/MemberBox.jsx [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+            add member
+          </Button>
+          <Button 
+            onClick={() => this.selectAll()} >
```
Comment (kamry-bowman) [inline app/src/components/MemberBox.jsx:195]: same, no need for an arrow function if you're not passing in an argument

File: app/src/components/MemberBox.jsx [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+            select all
+          </Button>
+          <Button
+            onClick={() => this.selectNone()} >
```
Comment (kamry-bowman) [inline app/src/components/MemberBox.jsx:199]: same

File: app/src/components/MemberBox.jsx [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+        phone: '',
+      })
+    }
+  }
+
+  selectAll = () => {
+    let updatedMembers = [...this.state.members]
+    updatedMembers.map(member => member.isChecked = true)
+
+    let checkboxes = document.getElementsByName('checkbox');
```
Comment (kamry-bowman) [inline app/src/components/MemberBox.jsx:152]: The main idea with React is that you just manage the data, and you tell React how to render based on the data in the `render` function. You want to be declarative about how things should render. What you usually don't want to do is be "imperative", where you're manipulating the DOM yourself.  See if you can manage this form without any DOM manipulation side effects in your methods

**Review by kamry-bowman — APPROVED** [review]
Great progress with this. I'm approving it, but I think you should go back and refactor this without any document calls at all if possible. You should just use props/state and the render method to cause changes in the DOM as much as possible. Happy to work with you on a zoom for it.

File: app/src/components/MemberBox.jsx (hunk 3 above)
Comment (guillearria) [reply app/src/components/MemberBox.jsx:109]: I love these comments, thank you. I implemented this and its functional. Super clean. Minor change to toggleMemberForm (missing parenthesis):

```
  toggleMemberForm = () => {
    this.setState(prevState => ({ 
      memberFormToggle: !prevState.memberFormToggle
    }) );
  }
```

File: app/src/components/MemberBox.jsx (hunk 4 above)
Comment (guillearria) [reply app/src/components/MemberBox.jsx:125]:
```
const formError = (toggleFlag) => {
  return ({
    color: 'red',
    fontSize: '0.9rem',
    display: toggleFlag ? 'block' : 'none',
  })
}

...

this.state = {
    members: [],
    name: '',
    phone: '',
    memberFormToggle: false,
    formErrorToggle: false,
}

...

if (this.state.name === "" || this.state.phone === "") { 
     this.setState({ formErrorToggle: true })    
} else { 
     this.setState({ formErrorToggle: false })
```

File: app/src/components/MemberBox.jsx (hunk 5 above)
Comment (guillearria) [reply app/src/components/MemberBox.jsx:132]: https://stackoverflow.com/questions/35791767/html-input-type-number-still-returning-a-string-when-accessed-from-javascript

File: app/src/components/MemberBox.jsx (hunk 9 above)
Comment (guillearria) [reply app/src/components/MemberBox.jsx:152]: Thank you Kamry, your suggestions have really helped me grasp the concept of thinking in terms of the state first. I've refactored all DOM manipulation except: 

```
    let checkboxes = document.getElementsByName('checkbox');
    for(var i=0, n=checkboxes.length;i<n;i++) {
      checkboxes[i].checked = bool;
    }
```

I'm stumped on how to deal with that one but it's minor. Way too tired right now lol.

**Pull request merged**